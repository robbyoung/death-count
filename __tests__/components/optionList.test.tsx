import React from 'react';
import renderer from 'react-test-renderer';
import OptionList from '../../app/components/optionList';

describe('Options List Component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<OptionList
            onSelect={() => undefined}
            options={['one', 'two', 'three', 'four']}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});