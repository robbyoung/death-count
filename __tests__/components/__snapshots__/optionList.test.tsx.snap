// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options List Component can render a non-selectable list 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#e84641",
        "borderBottomColor": "#592a29",
        "borderBottomWidth": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "margin": 10,
        }
      }
    >
      one
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#e84641",
        "borderBottomColor": "#592a29",
        "borderBottomWidth": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "margin": 10,
        }
      }
    >
      two
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#e84641",
        "borderBottomColor": "#592a29",
        "borderBottomWidth": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "margin": 10,
        }
      }
    >
      three
    </Text>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#e84641",
        "borderBottomColor": "#592a29",
        "borderBottomWidth": 2,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 22,
          "margin": 10,
        }
      }
    >
      four
    </Text>
  </View>,
]
`;

exports[`Options List Component can render a selectable list 1`] = `
Array [
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderBottomColor": "#592a29",
          "borderBottomWidth": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "margin": 10,
          }
        }
      >
        one
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderBottomColor": "#592a29",
          "borderBottomWidth": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "margin": 10,
          }
        }
      >
        two
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderBottomColor": "#592a29",
          "borderBottomWidth": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "margin": 10,
          }
        }
      >
        three
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#e84641",
          "borderBottomColor": "#592a29",
          "borderBottomWidth": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 22,
            "margin": 10,
          }
        }
      >
        four
      </Text>
    </View>
  </View>,
]
`;
